go/cmd/sqlflowserver/e2e_workflow_test.go

```diff
@@ -82,19 +82,21 @@ func TestEnd2EndWorkflow(t *testing.T) {
 		t.Fatalf("prepare test dataset failed: %v", err)
 	}
 
-	t.Run("CaseWorkflowTrainAndPredictDNNCustomImage", CaseWorkflowTrainAndPredictDNNCustomImage)
-	t.Run("CaseWorkflowTrainAndPredictDNN", CaseWorkflowTrainAndPredictDNN)
-	t.Run("CaseTrainDistributedPAIArgo", CaseTrainDistributedPAIArgo)
-	t.Run("CaseBackticksInSQL", CaseBackticksInSQL)
-	t.Run("CaseWorkflowStepErrorMessage", CaseWorkflowStepErrorMessage)
 	t.Run("CaseWorkflowRunBinary", caseWorkflowRunBinary)
 	t.Run("CaseWorkflowRunPythonScript", caseWorkflowRunPythonScript)
-	// test experimental workflow generation
 	os.Setenv("SQLFLOW_WORKFLOW_BACKEND", "experimental")
-	t.Run("CaseWorkflowTrainXgboost", CaseWorkflowTrainXgboost)
-	t.Run("CaseWorkflowTrainTensorFlow", caseWorkflowTrainTensorFlow)
-	t.Run("CaseWorkflowOptimize", caseWorkflowOptimize)
-	os.Setenv("SQLFLOW_WORKFLOW_BACKEND", "")
+	for i := 0; i < 2; i++ {
```

Comment:
Why duplicate the running of the same set of cases?

Reply:
- When `i = 0`, the environment variable `SQLFLOW_WORKFLOW_BACKEND` is equal to `experimental`. Then it will run using the new refactorized codegen. Then `SQLFLOW_WORKFLOW_BACKEND` is set to `""` [here](https://github.com/sql-machine-learning/sqlflow/pull/2959/files#diff-96e14ec8bafcb22364c3cc2f2648ba9cR98).
- When `i = 1`, `SQLFLOW_WORKFLOW_BACKEND=""`, so the tests will run using the old codegen.